frontend/packages/@depmap/data-explorer-2/src/utils/misc.ts

```diff
@@ -91,6 +95,34 @@ export const isSampleType = (dimensionType: string | null | undefined) => {
   );
 };
 
+export function convertDimensionToSliceId(
+  dimension: Partial<DataExplorerPlotConfigDimension>
+) {
+  if (!isCompleteDimension(dimension)) {
+    return null;
+  }
+
+  if (dimension.axis_type !== "raw_slice") {
+    throw new Error("Cannot convert a context to a slice ID!");
+  }
+
+  if (isSampleType(dimension.slice_type)) {
+    throw new Error(
+      "Cannot convert a sample to a slice ID! Only features are supported."
+    );
+  }
+
+  const expr = dimension.context.expr as { "==": [object, string] };
+  const feature = expr["=="][1];
+
+  return [
+    "slice",
+    urlLibEncode(dimension.dataset_id),
+    urlLibEncode(feature),
+    "label",
+  ].join("/");
```

Comment:
@snwessel This is where I construct a legacy slice ID. The public branch is pretty far behind master so this will probably be prone to the same bug you just fixed with Breadbox datasets. I think I can handle that here instead. What format should I use for that? `"breadbox/<dataset-id>/<feature>/label"` ?

Reply:
Oh wait, that's only relevant to custom analysis, right? This is just in Cell Line Selector that doesn't use any of those endpoints.

Reply:
Yeah, that bug was specific to custom analysis which uses some older endpoints that don't fully use the `data_access` interface. I'm not super familiar with how custom analysis works though, so it might be good to test these changes with a breadbox dataset before deploying.